includes/class-sensei-quiz.php

```diff
@@ -35,10 +35,10 @@ public function __construct( $file = __FILE__ ) {
 		add_action( 'template_redirect', array( $this, 'reset_button_click_listener' ) );
 
 		// Fire the complete quiz button submit for grading action.
-		add_action( 'sensei_single_quiz_content_inside_before', array( $this, 'user_quiz_submit_listener' ) );
+		add_action( 'template_redirect', array( $this, 'user_quiz_submit_listener' ) );
```

Comment:
Can we do this without changing the action? This would be a breaking change for any customizations that remove or modify these hooks.

Reply:
Hm, changing the action to what's better aligned with the callbacks' purpose is sort of the whole point here.. :thinking: 

There's also prior art right around these `add_action` calls that use `template_redirect`, so we also achieve better systemic alignment here.

Breaking change: yes - what is Sensei's policy for versioning? Such is usually handled by new major versions, but I'm aware WP and WC don't follow semver.

It seems like this change is fairly well "updatable" in customizations, if we highlight it well enough in release documentation.

Your thoughts?

Reply:
We have this [deprecation policy](https://senseilms.com/documentation/deprecation-policy/) that we try to adhere to, but it doesn't cover changing actions like this 🤔 

I was thinking that we might be able to fix the problem (refresh causing quiz submit) without changing the hook, but I agree that moving the functionality out of the rendering actions ultimately makes more sense.

Let me circle back on this.